docs/data/sources/google-tag-manager-client.md

```diff
@@ -3,17 +3,256 @@ title: Google Tag Manager Template - Amplitude Analytics Browser SDK
+--8<-- "includes/sdk-ts-browser/shared-configurations.md"
+- `Set configuration manually`. By selecting the corresponding option. The following configurations are the available ones with the value type in GTM template. 
+???config "GTM Configuration Options"
+    | <div class="big-column">Name</div>  | Description | Default Value |
+    | --- | --- | --- |
+    |`flushIntervalMillis`| `number`. The amount of time waiting to upload the event to the server in millionseconds. | 1 second.|
+    |`flushQueueSize`| `number`. The maximum number of events that can be stored locally before forcing an upload.  | 30 events. |
+    |`flushMaxRetries`| `number`. The max retry limits. | 5 times.|
+    |`logLevel`| `LogLevel.None` or `LogLevel.Error` or `LogLevel.Warn` or `LogLevel.Verbose` or `LogLevel.Debug`. The log level. | `LogLevel.Warn` |
+    |`minIdLength`| `number`. Overrides the minimum length of `user_id` & `device_id` fields. | `5` |
+    |`optOut`| `boolean`. If `optOut` is `true`, the event isn't sent to Amplitude's servers. | `false` |
+    |`serverUrl`| `string`. The server url events upload to. | `https://api2.amplitude.com/2/httpapi` | 
+    |`useBatch`| `boolean`. When `true`, uses the Batch API instead of the HTTP V2 API.| `false` |
+    |`appVersion`| `string`. The current version of your application. For example: "1.0.0" | `null` |
+    |`deviceId`| `string`. A device-specific identifier. | `UUID()` |
+    |`cookieExpiration`| `number`. The days when the cookie expires. | 365 days. |
+    |`cookieSameSite`| `string`. The SameSite attribute of the Set-Cookie HTTP response header. | `LAX` |
+    |`cookieSecure`| `boolean`. If restrict access to cookies or not. A cookie with the Secure attribute is only sent to the server with an encrypted request over the HTTPS protocol. | `false` |
+    |`disableCookies`| `boolean`. If disable cookies or not. If cookies is disable, using LocalStorage or MemoryStorage. | The cookies is enable by default. |
+    |`domain`| `string`. Set the top level domain. | `null` |
+    |`partnerId`| `string`. The partner Id value. Amplitude requires the customer who built an event ingestion integration to add the partner identifier to `partner_id`. | `null` |
+    |`sessionTimeout`| `number`. How long one session expire. | `30` minutes. |
+    |`userId`| `number`. ID for the user. Must have a minimum length of 5 characters unless overridden with the `min_user_length` option. | `undefined` |
+    |`trackingOptions`| `TrackingOptions`. Please check the `Optional tracking` section for more tracking options configuration. | Enable all tracking options by default. |
+    |`transport`| `TransportType.XHR` or `TransportType.SendBeacon` or `TransportType.Fetch`. Set the transport type. | `TransportType.Fetch` |
+- `Select a {{GTM variable}} from the list`. It's necessary to return an object containing the key-value pairs you wish to use for instance configuration. Ensure that the keys are part of the available configurations.
```

Comment:
👍 Didn't know how to use `{{GTM variable}}` to configure before. 

![image](https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/assets/31029607/71ba0d92-60ca-4d89-8206-97591597b7eb)

Curious if `{{GTM variable}}`s in the dropdown are pre-defined? Can they be used out-of-box?

Reply:
No. Actually, I try to summarize the options except `Use default values` and `Set configuration manually`. The others are all GTM variable, isn't it?

Reply:
I removed the `{{}}`. Hope that's clearer.